packages/repository-json-schema/src/build-schema.ts

```diff
@@ -14,8 +14,16 @@ import {
 import {JSONSchema6 as JSONSchema} from 'json-schema';
 import {JSON_SCHEMA_KEY} from './keys';
 
-export interface JsonSchemaOptions {
+// tslint:disable-next-line:no-any
+export interface JsonSchemaOptions<T extends object = any> {
   includeRelations?: boolean;
+
+  /// List of properties to exclude from the schema
+  exclude?: (keyof T)[];
```

Comment:
This type construct allows TypeScript to verify that `exclude` is using only existing model properties.

Consider the following incorrect usage:

```ts
{
  schema: getModelSchemaRef(TodoList, {exclude: ['id', 'foobar']}),
}
```

The compiler prints the following error:

```
Argument of type '{ exclude: ("id" | "foobar")[]; }' is not assignable to parameter of type 'JsonSchemaOptions<TodoList>'.
  Types of property 'exclude' are incompatible.
    Type '("id" | "foobar")[]' is not assignable to type '("id" | "title" | "color" | "todos" | "image" | "getId" | "getIdObject" | "toJSON" | "toObject")[]'.
      Type '"id" | "foobar"' is not assignable to type '"id" | "title" | "color" | "todos" | "image" | "getId" | "getIdObject" | "toJSON" | "toObject"'.
        Type '"foobar"' is not assignable to type '"id" | "title" | "color" | "todos" | "image" | "getId" | "getIdObject" | "toJSON" | "toObject"'.
```